lib/opensusebasetest.pm

```diff
@@ -334,8 +335,22 @@ sub wait_boot {
             type_line_svirt "$testapi::password";
             type_line_svirt "systemctl is-active network", expect => 'active';
             type_line_svirt 'systemctl is-active sshd',    expect => 'active';
-            # make sure we can reach the SSH server in the SUT
-            type_string "for i in {1..7}; do nc -zv $virsh_guest 22 && break || (echo \"retry: \$i\" ; sleep 5; false); done;\n";
+
+            # make sure we can reach the SSH server in the SUT, try up to 1 min (12 * 5s)
+            my $retries = 12;
+            for my $i (0 .. $retries) {
+                die "The SSH Port in the SUT could not be reached within 1 minute, considering a product issue" if $i == $retries;
+                my $port       = 22;
+                my %ports      = (tcp => {$port => {},});
+                my $host_check = check_ports($virsh_guest, 5, \%ports);
+                if ($host_check->{tcp}{$port}{open} == 1) {
+                    record_info("ssh port open", "check for port $port on $virsh_guest successful");
+                    last;
+                }
+                else {
+                    record_info("ssh port closed", "check for port $port on $virsh_guest failed", result => 'fail');
```

Comment:
Don't we want to retry 12 times before failing here? As with `result => 'fail'` we will exit after first failure. Or does it work differently?

Reply:
works differently - `result => 'fail'` just draws a red border for record_info - though I'm not convinced if we want to have that

Reply:
and I actually think you should change this . Let's keep red color for real failures . When it is expected failure you shouldn't mix it with unexpected ones .